crates/cext/src/transpiler/passes/commutative_cancellation.rs

```diff
@@ -0,0 +1,131 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::exit_codes::ExitCode;
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::converters::dag_to_circuit;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_transpiler::commutation_checker::get_standard_commutation_checker;
+use qiskit_transpiler::passes::cancel_commutations;
+use qiskit_transpiler::target::Target;
+
+/// @ingroup QkTranspilerPasses
+/// Run the CommutativeCancellation transpiler pass on a circuit.
+///
+/// This pass cancels the redundant (self-adjoint) gates through commutation relations.
+///
+/// @param circuit A pointer to the circuit to run CommutativeCancellation on. This circuit
+/// pointer to will be updated with the modified circuit if the pass is able to remove any gates.
+/// @param target This pass will attempt to accumulate all Z rotations into either
+/// an RZ, P or U1 gate, depending on which is already used in the circuit. If none
+/// is present in the circuit, this (optional) target argument is used as fallback to 
+/// decide which gate to use. If none of RZ, P or U1 are in the circuit or the target, 
+/// single-qubit Z rotations will not be optimized.
```

Comment:
May I suggest something like this (so that we don't need to stick to the imho weird current behavior of the pass)?

```suggestion
/// @param target This pass will attempt to remove pairs of consecutive
/// self-inverse gates, and accumulate consecutive Z rotations and consecutive
/// X rotations into single rotation gates.
```

Reply:
This doesn't really describe the target argument though. We could put this before the params section though, I think that's a fine addition.